### PR TITLE
Move polling watcher to own package

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
Move polling watcher to own package

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

File: packages/polling-watcher/lib/index.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  constructor (paths, options) {
+    super()
+    this.#paths = paths
+    this.#interval = options?.interval ?? 300_000
+    const signal = options?.signal
+    if (signal instanceof AbortSignal) {
+      if (!signal.aborted) {
+        signal.addEventListener('abort', () => this.abort(), { once: true })
+      } else {
+        this.#state = 'aborted'
```
Comment (holgerkoser) [inline packages/polling-watcher/lib/index.js:34]: return if aborted

File: packages/polling-watcher/lib/index.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      try {
+        // eslint-disable-next-line security/detect-non-literal-fs-filename -- path is not user input
+        const createHashEntry = async path => [path, sha256(await fs.readFile(path))]
+        const hashEntries = await Promise.all(paths.map(createHashEntry))
+        this.#hashes = new Map(hashEntries)
+        if (this.#state === 'initial') {
+          this.#state = 'ready'
+          logger.info('[polling-watcher] watcher is ready')
+          this.emit('ready')
+        }
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply packages/polling-watcher/lib/index.js]: State management has been removed

File: packages/polling-watcher/lib/index.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  get interval () {
+    return this.#interval
+  }
+
+  get state () {
+    return this.#state
+  }
+
+  run (fn) {
+    if (['initial', 'ready'].includes(this.#state)) {
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply packages/polling-watcher/lib/index.js]: State management has been removed

File: packages/polling-watcher/lib/index.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      const tryReadFile = async path => {
+        try {
+          // eslint-disable-next-line security/detect-non-literal-fs-filename -- path is not user input
+          const data = await fs.readFile(path)
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply packages/polling-watcher/lib/index.js]: The error is catched and a log message is written.

File: packages/polling-watcher/lib/index.js (hunk 1 above)
Comment (holgerkoser) [reply packages/polling-watcher/lib/index.js]: state management has been removed

File: packages/polling-watcher/lib/index.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        signal.addEventListener('abort', () => this.abort(), { once: true })
+      } else {
+        this.#state = 'aborted'
+      }
+    }
+    const init = async () => {
+      try {
+        // eslint-disable-next-line security/detect-non-literal-fs-filename -- path is not user input
+        const createHashEntry = async path => [path, sha256(await fs.readFile(path))]
+        const hashEntries = await Promise.all(paths.map(createHashEntry))
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply packages/polling-watcher/lib/index.js]: The files must exist. Otherwise the application cannot start. The container manager ist responsible for restart or retry.

File: packages/polling-watcher/__tests__/polling-watcher.test.js [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+      await isReady(watcher)
+      expect(watcher.state).toBe('ready')
+      ac.abort()
+      expect(watcher.state).toBe('ready')
+      watcher.run(fn)
+      fs.readFile.mockResolvedValueOnce(newContent)
+      await jest.advanceTimersByTimeAsync(watcher.interval + 1)
+      expect(fn).toHaveBeenCalledTimes(1)
+      watcher.abort()
+      expect(watcher.state).toBe('aborted')
```
[the remark this answers is not in the document]
Comment (holgerkoser) [reply packages/polling-watcher/__tests__/polling-watcher.test.js]: ✔️

File: backend/package.json [hunk 7]
```diff
@@ -117,7 +117,7 @@
       "<rootDir>/jest.setup.js"
     ]
   },
-  "packageManager": "yarn@4.5.3",
```
Comment (petersutter) [inline backend/package.json:120]: revert these changes

**Review by grolu — APPROVED** [review]
/lgtm

**Review by petersutter — APPROVED** [review]
/lgtm
